### PR TITLE
fix precursor scan finding problems in difficult cases; better error messages

### PR DESCRIPTION
**Pull request by rmillikin**

### CODE REVIEW

File: Thermo/ThermoDynamicData.cs [hunk 1]
```diff
@@ -64,6 +71,11 @@ public static ThermoDynamicData InitiateDynamicConnection(string filePath, IFilt
 
             var thermoGlobalParams = GetAllGlobalStuff(_rawConnection, precursorInfoArray, filePath);
 
+            if (thermoGlobalParams.msOrderByScan.Length == 1 && thermoGlobalParams.msOrderByScan[0] == 0)
```
[the remark this answers is not in the document]
Comment (rmillikin) [reply Thermo/ThermoDynamicData.cs]: I'm not sure why the errored files are read this way, but they are. If there is only one scan and it's MS order is 0, it indicates an errored reading result

File: Thermo/ThermoFile.cs [hunk 2]
```diff
@@ -277,34 +277,35 @@ protected static ThermoGlobalParams GetAllGlobalStuff(IXRawfile5 _rawConnection,
                 // THIS METHOD IS BUGGY!!! DO NOT USE
                 //theConnection.FindPrecursorMassInFullScan(nScanNumber, ref pnMasterScan, ref pdFoundMass, ref pdHeaderMass, ref pnChargeState);
 
-                int oneBasedPrecursorScanNumber;
+                int oneBasedPrecursorScanNumber = -1;
```
[the remark this answers is not in the document]
Comment (rmillikin) [reply Thermo/ThermoFile.cs]: Not sure, I'll check it out

File: Thermo/ThermoDynamicData.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (acesnik) [reply Thermo/ThermoDynamicData.cs]: Okay, maybe make a comment for future eyes to see.

File: Thermo/ThermoFile.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (acesnik) [reply Thermo/ThermoFile.cs]: Okay, sounds like a good change if you're reducing the complexity as such!

[the remark this answers is not in the document]
Comment (rmillikin) [reply Thermo/ThermoFile.cs]: I'm not finding any references to FindPrecursorMassInFullScan() besides the commented-out line 278

**Pull request merged**